packages/kit/src/core/postbuild/analyse.js

```diff
@@ -52,97 +52,54 @@ async function analyse({ manifest_path, env }) {
 		routes: new Map()
 	};
 
-	// analyse nodes
-	for (const loader of manifest._.nodes) {
-		const node = await loader();
+	const nodes = await Promise.all(manifest._.nodes.map((loader) => loader()));
 
+	// analyse nodes
+	for (const node of nodes) {
 		metadata.nodes[node.index] = {
 			has_server_load: node.server?.load !== undefined || node.server?.trailingSlash !== undefined
 		};
 	}
 
 	// analyse routes
 	for (const route of manifest._.routes) {
-		/** @type {Array<'GET' | 'POST'>} */
-		const page_methods = [];
-
-		/** @type {(import('types').HttpMethod | '*')[]} */
-		const api_methods = [];
+		const page =
+			route.page &&
+			analyse_page(
+				route.page.layouts.map((n) => (n === undefined ? n : nodes[n])),
+				nodes[route.page.leaf]
+			);
 
-		/** @type {import('types').PrerenderOption | undefined} */
-		let prerender = undefined;
-		/** @type {any} */
-		let config = undefined;
-		/** @type {import('types').PrerenderEntryGenerator | undefined} */
-		let entries = undefined;
+		const endpoint = route.endpoint && analyse_endpoint(route, await route.endpoint());
 
-		if (route.endpoint) {
-			const mod = await route.endpoint();
-			if (mod.prerender !== undefined) {
-				validate_server_exports(mod, route.id);
+		if (page?.prerender && endpoint?.prerender) {
+			throw new Error(`Cannot prerender a route with both +page and +server files (${route.id})`);
+		}
 
-				if (mod.prerender && (mod.POST || mod.PATCH || mod.PUT || mod.DELETE)) {
+		if (page?.config && endpoint?.config) {
+			for (const key in { ...page.config, ...endpoint.config }) {
+				if (JSON.stringify(page.config[key]) !== JSON.stringify(endpoint.config[key])) {
```

Comment:
Is it guaranteed that the keys of both of these are in the same order? i.e. could it fail because content is the same but in different order?

Reply:
Below the top level, yes. I didn't think it was worth adding deep equality checking to handle that case, we're in super niche territory